diracx-db/src/diracx/db/sql/utils.py

```diff
@@ -212,10 +264,62 @@ async def ping(self):
             raise SQLDBUnavailable("Cannot ping the DB") from e
 
 
-def apply_search_filters(table, stmt, search):
-    # Apply any filters
+def find_time_resolution(value):
+    if isinstance(value, datetime):
+        return None, value
+    if match := re.fullmatch(
+        r"\d{4}(-\d{2}(-\d{2}(([ T])\d{2}(:\d{2}(:\d{2}(\.\d{6}Z?)?)?)?)?)?)?", value
+    ):
+        if match.group(6):
+            precision, pattern = "SECOND", r"\1-\2-\3 \4:\5:\6"
+        elif match.group(5):
+            precision, pattern = "MINUTE", r"\1-\2-\3 \4:\5"
+        elif match.group(3):
+            precision, pattern = "HOUR", r"\1-\2-\3 \4"
+        elif match.group(2):
+            precision, pattern = "DAY", r"\1-\2-\3"
+        elif match.group(1):
+            precision, pattern = "MONTH", r"\1-\2"
+        else:
+            precision, pattern = "YEAR", r"\1"
+        return (
+            precision,
+            re.sub(
+                r"^(\d{4})-?(\d{2})?-?(\d{2})?[ T]?(\d{2})?:?(\d{2})?:?(\d{2})?\.?(\d{6})?Z?$",
+                pattern,
+                value,
+            ),
+        )
+
+    raise InvalidQueryError(f"Cannot parse {value=}")
+
+
+def apply_search_filters(column_mapping, stmt, search):
     for query in search:
-        column = table.columns[query["parameter"]]
+        try:
+            column = column_mapping(query["parameter"])
+        except KeyError as e:
+            raise InvalidQueryError(f"Unknown column {query['parameter']}") from e
+
+        if isinstance(column.type, DateTime):
+            if "value" in query and isinstance(query["value"], str):
```

Comment:
What would be a search case without `value` ? :thinking:

Reply:
`parameter in values` (plural!)

Reply:
Ouh... that is actually super error prone no ? Should we use a single word for both  case, and even a constant for it ? something like `SEARCH_VALUE="value"` ?

Reply:
Well `"value"` is always a scalar and `"values"` is always a list of scalars.

The schema of the search body can be validated to make sure you only use the right key with the right kind of operator.